WISE-Community/WISE-Client
# Created Node class

## Changes

- Created Node class. It does not have all the fields and functions yet, but we can add them gradually.
- Moved ```ProjectService.getNodeIconByNodeId()``` function to ```Node.getIcon()``` and updated references.
- Created ```ProjectService.getNode(nodeId)```, which returns a Node instance. Note that this is different from ```this.ProjectService.getNodeByNodeId(nodeId)```, which returns a JSON object. We can gradually replace all calls with the new function.
- Added unit tests.

## Test

- Update node icons in the AT by adding/editing the icons JSON block for a step or group.
Here's a sample icons JSON block:
```
"icons": {
        "default": {
            "fontName": "info",
            "color": "#2196F3",
            "fontSet": "material-icons",
            "type": "font"
        }
    }
```

- Verify that node icons appear correctly in CM, AT, VLE.
- Verify that there is no noticeable degradation of performance.

#97
